src/mmrelay/meshtastic_utils.py

```
@@ -1199,8 +1242,34 @@ def connect_meshtastic(
+                            except concurrent.futures.TimeoutError as err:
+                                logger.error(
+                                    f"BLE connect() call timed out after 30 seconds for {ble_address}. "
+                                    "This may indicate a BlueZ or adapter issue."
+                                )
+                                # Don't use iface if connect() timed out - it may be in an inconsistent state
+                                iface = None
+                                raise TimeoutError(
+                                    f"BLE connect() timed out for {ble_address}. "
+                                    "BlueZ may be in a bad state. Try: "
+                                    "'sudo systemctl restart bluetooth' or 'bluetoothctl disconnect {ble_address}'"
+                                ) from err
```

Ruff (0.14.11)

Avoid specifying long messages outside the exception class

(TRY003)